src/Estimators/src/UkfState.cpp

```diff
@@ -0,0 +1,543 @@
+
+                subModelJointMotorTorque[subModelIdx](jointIdx)
+                    = currentState[stateVariableHandler.getVariable("tau_m").offset
+                                   + subModelList[subModelIdx].getJointMapping()[jointIdx]];
+
+                subModelFrictionTorque[subModelIdx](jointIdx)
+                    = currentState[stateVariableHandler.getVariable("tau_F").offset
+                                   + subModelList[subModelIdx].getJointMapping()[jointIdx]];
+            }
+
+            for (auto& [key, value] : subModelList[subModelIdx].getFTList())
+            {
+                FTMap[key] = currentState.segment(stateVariableHandler.getVariable(key).offset,
+                                                  stateVariableHandler.getVariable(key).size);
+            }
+
+            for (int idx = 0; idx < subModelList[subModelIdx].getNrOfExternalContact(); idx++)
+            {
+                extContactMap[subModelList[subModelIdx].getExternalContact(idx)]
+                    = currentState
+                          .segment(stateVariableHandler
+                                       .getVariable(
+                                           subModelList[subModelIdx].getExternalContact(idx))
+                                       .offset,
+                                   stateVariableHandler
+                                       .getVariable(
+                                           subModelList[subModelIdx].getExternalContact(idx))
+                                       .size);
+            }
+        }
+    }
+
+    bool updateState()
+    {
+        // Update kindyn full model
+        kinDynFullModel->setRobotState(ukfInput.robotBasePose.transform(),
+                                       ukfInput.robotJointPositions,
+                                       iDynTree::make_span(ukfInput.robotBaseVelocity.data(),
+                                                           manif::SE3d::Tangent::DoF),
+                                       jointVelocityState,
+                                       gravity);
+
+        // compute joint acceleration per each sub-model
+        for (int subModelIdx = 0; subModelIdx < subModelList.size(); subModelIdx++)
+        {
+            // Update the kindyn wrapper object of the submodel
+            kinDynWrapperList[subModelIdx]->updateState(ukfInput.robotBaseAcceleration,
+                                                        jointAccelerationState,
+                                                        UpdateMode::RobotDynamicsOnly);
+
+            if (subModelList[subModelIdx].getModel().getNrOfDOFs() > 0)
+            {
+                totalTorqueFromContacts[subModelIdx].setZero();
+
+                // Contribution of FT measurements
+                for (auto& [key, value] : subModelList[subModelIdx].getFTList())
+                {
+                    wrench = static_cast<int>(value.forceDirection) * FTMap[key].array();
```

Comment:
Probably this is going to work
```suggestion
                    wrench.noalias() = static_cast<int>(value.forceDirection) * FTMap[key];
```